### PR TITLE
feat: add courseware format

### PR DESCRIPTION
**Pull request by asadali145**
# What are the relevant tickets?

Fixes https://github.com/mitodl/mitxpro/issues/2611

# Description (What does it do?)

Adds courseware format added in https://github.com/mitodl/mitxpro/issues/2611.

# How can this be tested?

Run the following commands:

- `dbt build -s +int__mitxpro__courses --vars 'schema_suffix: asad'`
- `dbt build -s +int__mitxpro__programs --vars 'schema_suffix: asad' `
- `dbt build -s +stg__mitxpro__app__postgres__cms_programpage --vars 'schema_suffix: asad'`
- `dbt build -s +stg__mitxpro__app__postgres__cms_coursepage --vars 'schema_suffix: asad'`
- `dbt build -s +marts__mitxpro_ecommerce_productlist --vars 'schema_suffix: asad'`

### CODE REVIEW

File: src/ol_dbt/models/staging/mitxpro/_stg_mitxpro__models.yml [hunk 1]
```diff
@@ -1525,6 +1525,8 @@ models:
   - name: cms_coursepage_duration
     description: str, a short description indicating how long the course takes to
       complete (e.g. '4 weeks')
+  - name: cms_coursepage_format
+    description: str, format of the course (Online, Other)
```
Comment (rachellougee) [inline src/ol_dbt/models/staging/mitxpro/_stg_mitxpro__models.yml:1529]: Will there be blank? if not, then I would suggest
```suggestion
    description: str, format of the program or course (Online, Other)
    tests:
    - not_null
    - accepted_values:
        values: ['Online', 'Other']
```

File: src/ol_dbt/models/marts/mitxpro/_marts__mitxpro__models.yml [hunk 2]
```diff
@@ -42,6 +42,8 @@ models:
   - name: duration
     description: str, a short description indicating how long the course or program
       takes to complete (e.g. '4 weeks')
+  - name: format
```
Comment (rachellougee) [inline src/ol_dbt/models/marts/mitxpro/_marts__mitxpro__models.yml:45]: This name doesn't match with what it's added in the mart

File: src/ol_dbt/models/marts/mitxpro/marts__mitxpro_ecommerce_productlist.sql [hunk 3]
```diff
@@ -80,6 +80,7 @@ select
     ) as link
     , concat(programs.program_readable_id, '+', course_runs.courserun_tag) as product_parent_run_id
     , courses.cms_coursepage_duration as duration
+    , courses.cms_coursepage_format as course_format
```
Comment (rachellougee) [inline src/ol_dbt/models/marts/mitxpro/marts__mitxpro_ecommerce_productlist.sql:83]: @KatelynGit I assume you need this field in the mart?

**Review by rachellougee — CHANGES_REQUESTED** [review]
I left a couple of comments. Also, it doesn't seem like this two fields are in our data platform yet? We will need to add them in the raw before merging.

**Comment by asadali145** [comment]
Review changes resolved in 767c7f290816c9e491dc3a4bc2dfa28c0102c466.

**Review by rachellougee — APPROVED** [review]
Code looks good. But I would suggest making sure fields exist in raw tables on production and run tests before merging as I still don't see `format` added to these 4 raw tables.

**Comment by asadali145** [comment]
@rachellougee I have tried to test this PR multiple times but I am getting an error that I wasn't getting when I created this PR. Do you know any workarounds for this issue?
I have tried:

- `dbt build -s +int__mitxpro__courses --vars 'schema_suffix: asad' --target dev_production`
- `dbt build -s +int__mitxpro__courses --vars 'schema_suffix: asad'`

Screen Shot 2023-12-05 at 1 36 22 PM

**Comment by rachellougee** [comment]
@asadali145 The error you are getting are unrelated to your PR but it's related to relationships tests in [this line](https://github.com/mitodl/ol-data-platform/blob/main/src/ol_dbt/models/intermediate/mitxpro/_int_mitxpro__models.yml#L1330), that test has been always there.  So `dbt build -s +int__mitxpro__courses` will run all the tests related to this model, but it doesn't seem to run `dbt run ` on all the related models used in the test. In this case, you have never ran `int__mitxpro__courserun_certificates` and 
`int__mitxpro__courserun_grades` under your schema

To workaround that, you can run 
- these two separate commands since you only need to test stage and intermediate models that you modified
```
dbt build --select stg__mitxpro__app__postgres__cms_coursepage --vars 'schema_suffix: asad' --target dev_production
dbt build --select  int__mitxpro__courses --vars 'schema_suffix: asad' --target dev_production
```
- `dbt build --select @int__mitxpro__courses --vars 'schema_suffix: asad' --target dev_production`. But this will run the parents of the children of the courses model, which takes long time and may not be necessary https://docs.getdbt.com/reference/node-selection/graph-operators

**Comment by rachellougee** [comment]
@asadali145 Also, I just ran this PR changes under my schema. It works fine, so I think you can merge it now.

**Comment by asadali145** [comment]
@rachellougee Thanks for your help. Merging this PR.

**Pull request merged**